src/sentry/issues/escalating.py

```diff
@@ -343,20 +343,25 @@ def manage_issue_states(
             add_group_to_inbox(group, GroupInboxReason.ONGOING, snooze_details)
             record_group_history(group, GroupHistoryStatus.ONGOING)
     elif group_inbox_reason == GroupInboxReason.UNIGNORED:
-        updated = Group.objects.filter(
+        groups = Group.objects.filter(
             id=group.id, status__in=[GroupStatus.RESOLVED, GroupStatus.IGNORED]
-        ).update(status=GroupStatus.UNRESOLVED, substatus=GroupSubStatus.ONGOING)
-        if updated:
-            group.status = GroupStatus.UNRESOLVED
-            group.substatus = GroupSubStatus.ONGOING
-            post_save.send(
-                sender=Group,
-                instance=group,
-                created=False,
-                update_fields=["status", "substatus"],
-            )
-            add_group_to_inbox(group, GroupInboxReason.UNIGNORED, snooze_details)
-            record_group_history(group, GroupHistoryStatus.UNIGNORED)
+        )
```

Comment:
I'm not sure it would be possible to get multiple groups when we're filtering on `group.id`?

Reply:
Changed the logic to grab the first entry from the `.filter`.

Reply:
Could we use a conditional update (https://docs.djangoproject.com/en/4.2/ref/models/conditional-expressions/#conditional-update) here instead of retrieving the first? 

The reason why we wanna do the update chained with the filter is we want the resulting query to run all in one shot in postgres. I'm not too sure how likely it is to run into race conditions for updating the `status` and `substatus` values for the same group but if we break them up its possible to update the state when it was previously updated already.

Reply:
Hmm, now that I think about it if we do a conditional update, we can't retrieve the `substatus` field value at all and won't be able to call `add_group_to_inbox` appropriately. 

I suppose an alternative approach is we break this up into two parts:
1. 
```
ongoing_set = Group.objects.filter(
            id=group.id, status__in=[GroupStatus.RESOLVED, GroupStatus.IGNORED]
        ).update(
            status=GroupStatus.UNRESOLVED,
            substatus=Case(
                When(
                    first_seen__lt=datetime.now(timezone.utc) - timedelta(days=3),
                    then=Value(GroupStatus.ONGOING),
                ),
                default=F('substatus'),
            ),
        )
```
2. 
```
new_set = Group.objects.filter(
            id=group.id, status__in=[GroupStatus.RESOLVED, GroupStatus.IGNORED]
        ).update(
            status=GroupStatus.UNRESOLVED,
            substatus=Case(
                When(
                    first_seen__gte=datetime.now(timezone.utc) - timedelta(days=3),
                    then=Value(GroupStatus.NEW),
                ),
                default=F('substatus'),
            ),
        )
```

Then fire the appropriate methods based on:
```
updated = new_set or ongoing_set

if not ongoing_set and not new_set: # this group isn't resolved or ongoing, skip it
  return

if new_set:
  # New things

if ongoing_set:
  # ongoing things

```